pkg/dash/workflow.go

```diff
@@ -186,6 +189,52 @@ func Watch(config Config, watchDir, jsonnetFile string, targets *[]string) error
 	<-done
 	return nil
 }
+
+// Export renders Jsonnet dashboards then saves them to a directory
+func Export(config Config, jsonnetFile, dashboardDir string, targets *[]string) error {
+	folderID, err := folderId(config, jsonnetFile)
+	if err != nil {
+		var fID int64 = 0
+		folderID = &fID
+		fmt.Println("Folder not found and/or configured. Applying to \"General\" folder.")
+	}
+	boards, err := renderDashboards(jsonnetFile, targets, *folderID)
+	if err != nil {
+		return err
+	}
+
+	for name, board := range boards {
+		boardJSON, err := board.GetDashboardJSON()
+		if err != nil {
+			return err
+		}
+		boardPath := path.Join(dashboardDir, board.Name)
+		if !strings.HasSuffix(board.Name, ".json") {
+			boardPath += ".json"
+		}
+		existingBoardJSONBytes, err := ioutil.ReadFile(boardPath)
+		existingBoardJSON := string(existingBoardJSONBytes)
+
+		if err != nil && !os.IsNotExist(err) {
+			fmt.Println(err)
+			return err
+		}
+
+		if os.IsNotExist(err) {
+			fmt.Println(name, green("added"))
+		} else if boardJSON == existingBoardJSON {
+			fmt.Println(name, yellow("unchanged"))
+		} else {
+			fmt.Println(name, green("updated"))
+		}
+		err = ioutil.WriteFile(boardPath, []byte(boardJSON), 0644)
+		if err != nil {
+			return err
+		}
```

Comment:
I think `ioutil.WriteFile` should go above the `fmt.Println` messages. If an error occurs, the message will have been incorrect. For example if the output directory doesn't exist:

```
▸ grr export chris.trott/dashboards.jsonnet test
rc-mem-reqs added
2020/06/06 19:35:12 open test/rc-mem-reqs.json: no such file or directory
```

Reply:
yeah, statements are in the past tense, so yes.